### ISSUE TITLE
A3C loss functions

### ISSUE THREAD

**Issue by gabrieledcjr** [awjuliani/DeepRL-Agents#11]
In A3C,

```
self.value_loss = 0.5 * tf.reduce_sum(tf.square(self.target_v - tf.reshape(self.value,[-1])))
self.entropy = - tf.reduce_sum(self.policy * tf.log(self.policy))
self.policy_loss = -tf.reduce_sum(tf.log(self.responsible_outputs)*self.advantages)
self.loss = 0.5 * self.value_loss + self.policy_loss - self.entropy * 0.01
```

since, 0.5 is already multiplied in self.value_loss, should 

`self.loss = self.value_loss + self.policy_loss - self.entropy * 0.01`

**Comment by awjuliani** [comment]
Hi Gabriel, this is actually a feature, in a strange way. In A3C the value loss is often scaled by some coefficient in order to ensure smaller (and hopefully more stable) changes to the value function. The first 0.5 is part of the value loss itself, while the 0.5 in the full loss function is the regularizer parameter, similar to the 0.01 the entropy is multiplied by. I have actually seen the value loss scaled down to as much as 0.1 in some cases. In practice the separation can seem somewhat arbitrary though.
